src/GraphQL.Tests/Bugs/Issue2932_DemoDIGraphType.cs

```diff
@@ -0,0 +1,209 @@
+            // also demonstrates auto inference of return type and removal of "Async" from asynchronous methods
+            public Task<IEnumerable<string>> ChildrenAsync() => _service1.GetChildrenAsync(Source.Id, RequestAborted);
+
+            // sample of a static method, pulling in a specific service
+            public static int Service2Test([FromServices] Service2 service2) => service2.Value;
+
+            // sample of returning data from a scoped service
+            public int Counter() => _service1.Count;
+
+            // sample of the field resolver creating a scope for this method, and returning data from a scoped service within that scope
+            [Scoped]
+            public int ScopedCounter() => _service1.Count;
+        }
+
+        public class DIGraphType<DIObject, TSourceType> : AutoRegisteringObjectGraphType<TSourceType>
+            where DIObject : DIGraph<TSourceType>
+        {
+            protected override void ConfigureGraph()
+            {
+                // do not configure attributes set on TSourceType
+                // base.ConfigureGraph();
+
+                // configure attributes set on DIObject instead
+                // todo: AutoRegisteringHelper.ApplyGraphQLAttributes should be public for this sample to work as written
+                AutoRegisteringHelper.ApplyGraphQLAttributes<DIObject>(this);
+            }
+
+            // only process methods declared directly on DIObject -- not anything declared on TSourceType
+            protected override IEnumerable<MemberInfo> GetRegisteredMembers()
+            {
+                return typeof(DIObject)
+                    .GetMethods(BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.DeclaredOnly);
+            }
+
+            // each field resolver will build a new instance of DIObject
+            protected override LambdaExpression BuildMemberInstanceExpression(MemberInfo memberInfo)
+                => (Expression<Func<IResolveFieldContext, DIObject>>)(context => MemberInstanceFunc(context));
+
+            private DIObject MemberInstanceFunc(IResolveFieldContext context)
+            {
+                // create a new instance of DIObject, filling in any constructor arguments from DI
+                var graph = ActivatorUtilities.CreateInstance<DIObject>(context.RequestServices ?? throw new MissingRequestServicesException());
+                // set the context
+                graph.Context = context;
+                // return the object
+                return graph;
+            }
+        }
+
+        public class DIGraph<TSourceType>
+        {
+            public IResolveFieldContext Context { get; internal set; } = null!;
+            public TSourceType Source => (TSourceType)Context.Source!;
+            public CancellationToken RequestAborted => Context.CancellationToken;
+        }
+
+        [AttributeUsage(AttributeTargets.Parameter)]
+        public class FromSourceAttribute : GraphQLAttribute
```

Comment:
I can, but it doesn’t really serve a purpose under normal use, because the method is written on the instance of the source. In another way of speaking “this” always points to the source so the attribute is pointless. It’s only needed here because of the way AutoRegisteringObjectGraphType is adapted to fit another use case. 

Lmk if you still think I should add it.